src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -96,8 +122,7 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
                 return;
             }
 
-            var semanticModel = syntaxContext.SemanticModel;
-            if (semanticModel.GetSymbolInfo(notEquals).GetAnySymbol().IsUserDefinedOperator())
+            if (semanticModel.GetSymbolInfo(comparison).GetAnySymbol().IsUserDefinedOperator())
```

Comment:
💭 This will rule out the feature for many types. We may be able to improve the experience by allowing user-defined operators, but ensuring a single Fix All operation only corrects operations that use the same operator. For example:

* If all uses of `== null` and `!= null` for a variable use the default operator, a Fix All on the variable will fix all cases where no user-defined operators are involved
* If one or more uses of `==` or `!=` involve a user-defined operator for a variable with the declared type `T`, then a Fix All on that variable will fix all instances involving variables with declared type `T`, whether or not they use user-defined operators. Alternately, this scenario could fix all instances involving variables with declared type `T` which use one or more user-defined operators.

The equivalence ID of the code fix can be used to "bucket" fixes according to the above conditions.

If you like the idea but aren't ready to implement it, feel free to leave this as a follow-up issue by filing a new bug referencing this comment.

Reply:
This is not part of this PR, the only diff here is `notEquals` -> `comparison`

Reply:
Filed https://github.com/dotnet/roslyn/issues/23374